### PR TITLE
Bench Pretty Printer support for hackmd format

### LINKED ISSUE #1614: Correct pretty printer to print Markdown format
Currently, when the pretty printer is used for benchmarks, the format isn't completely compatible with the markdown format.
The goal is to correct the pretty printer to fit this requirement.

### PR DESCRIPTION
**Pull request by maiste**
Change the printer of `trace_stat` to export the text as a format understood by the `hackmd`.

Closes #1614

### CODE REVIEW

File: bench/irmin-pack/trace_stat_summary_pp.ml [hunk 1]
```diff
@@ -1006,17 +1018,24 @@ let unsafe_pp sample_count ppf summary_names (summaries : Summary.t list) =
     |> PrintBox_text.to_string
   in
   let table1 =
-    let only_one_summary = List.length summaries = 1 in
+    let summart_length = List.length summaries in
```
Comment (CraigFe) [inline bench/irmin-pack/trace_stat_summary_pp.ml:1021]:
```suggestion
    let summary_length = List.length summaries in
```

**Review by CraigFe — COMMENTED** [review]
LGTM.

Might be nice to have an expect test of this format (now that we want to try to keep it compliant with something), but I don't think this needs to come as part of this PR.

**Comment by maiste** [comment]
Thanks for the review :+1: 

I made the correction.

As we discuss offline, I don't think it worth adding a test for the format as we are not sure to keep it that way.

**Comment by Ngoguey42** [comment]
Have you checked that this still works when only one summary is passed to pp?

**Comment by maiste** [comment]
> Have you checked that this still works when only one summary is passed to pp?

Indeed, I haven't checked and it seems that it's not printable. I'm going to push a correction.

File: bench/irmin-pack/trace_stat_summary_pp.ml [hunk 2]
```diff
@@ -155,8 +160,10 @@ module Table0 = struct
             | `V0 -> "v0"))
 
   let box_of_summaries_config summary_names (summaries : summary list) =
-    let row0 =
-      if List.length summary_names = 1 then [] else [ "" :: summary_names ]
+    let summary_name_length = List.length summary_names in
+    let row0 = [ "" :: summary_names ] in
```
[the remark this answers is not in the document]
Comment (Ngoguey42) [reply bench/irmin-pack/trace_stat_summary_pp.ml:164]: I don't remember exactly what this piece of code does. As long as the pp works on both the single-summary case and the multi-summary case, I trust your judgement :)

[the remark this answers is not in the document]
Comment (maiste) [reply bench/irmin-pack/trace_stat_summary_pp.ml:164]: The idea before the modification was to don't print a header if you have one summary.

Without my change:
```
| Hostname   | AwesomeHostname | 
```

With my change (markdown compatible):
```
|                  | Summary_name
|--                | --
| Hostname         | AwesomeHostname
```

In markdown it is printed this way:
|                  | Summary_name
|--                | --
| Hostname         | AwesomeHostname

If it's ok for you, I keep it this way :+1:

[the remark this answers is not in the document]
Comment (Ngoguey42) [reply bench/irmin-pack/trace_stat_summary_pp.ml:164]: Thanks!

**Review by Ngoguey42 — APPROVED** [review]
LGTM. Can be merged as soon as CI is happy

**Comment by maiste** [comment]
Sure! Thanks for the review :+1:

**Pull request merged**